pkg/repo/index.go

```diff
@@ -371,11 +375,10 @@ func loadIndex(data []byte, source string) (*IndexFile, error) {
 				cvs = append(cvs[:idx], cvs[idx+1:]...)
 			}
 		}
+		// adjust slice to only contain a set of valid versions
+		i.Entries[name] = cvs
 	}
 	i.SortEntries()
-	if i.APIVersion == "" {
```

Comment:
I recommend not combining unrelated change into a single PR. Agree, I see no reason not to "optimize" the behavior here. But, IMHO coupling unrelated changes (the bugfix for i.Entries filtering, and this optimization) is bad practice.

Reply:
Hi, I agree. Will be able to change it in a couple of days as I currently out of town.

Reply:
reverted the change regarding the check if APIVersion is empty